Feature/db schema-diagram

RileyManda opened the pull request:
### Medical tables
We @ClarenceAtim AND @RileyManda  have completed the implementation of  database schema according to the stipulated [requirements ](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/images/clinic_diagram.png): 🌱 

- [x] Created patient table
- [x] Created medical history table 
- [x] Created indexes and foreign key medicalhistories and  history table

### Invoices|Invoice-Items|Treatments

- [x] Created invoices table
- [x] Created invoice_items table
- [x] Created invoice_items and invoice indexes
- [x] Created indexes and foreign keys 

Please kindly review our implementation and suggest any improvements and changes.

Thank you 🙏🏻

RileyManda committed:
Created db_schema_diagram file

RileyManda committed:
Created database clinic

RileyManda committed:
Included animals schema image to project|created invoice and invoice-items tables and indexes

RileyManda committed:
Implemented fk and indexes

RileyManda committed:
Implemented a one to many relationship treatments and incoice_items

RileyManda committed:
Treatments to invoice indexes

RileyManda committed:
Typo error fix on table name

ClarenceAtim committed:
Created the table for patient, medical history

RileyManda committed:
Updated authors anf features-readme

ClarenceAtim committed:
Created te fk and indice for medical history

LionRouge1 commented on db_schema_diagram.sql:
```diff
@@ -0,0 +1,51 @@
+  date_of_birth DATE NOT NULL
+);
+
+CREATE TABLE medical_histories (
+  id SERIAL PRIMARY KEY,
+  admitted_at TIMESTAMP,
+  patient_id INT NOT NULL,
+  status VARCHAR(150)
+);
+CREATE TABLE treatments (
+    id SERIAL PRIMARY KEY,
+    type VARCHAR(255),
+    NAME VARCHAR(255)
+);
```
- [ ] Kindly note that there is a many-to-many relationship between the `medical_histories` and `treatments` tables. Therefore it would be nice to have a **JOIN TABLE** `medical_histories_treatments` reflecting the **many-to-many** relationship with `medical_history_id` and `treatment_id` as foreign keys referencing the `medical_histories` and `treatments` tables respectively

LionRouge1 commented on db_schema_diagram.sql:
```diff
@@ -0,0 +1,51 @@
+FOREIGN KEY (patient_id) REFERENCES patients(id) ON DELETE CASCADE;
+-- invoice index keys 
+
+CREATE INDEX idx_invoice_items_invoice_id ON invoice_items(invoice_id);
+CREATE INDEX idx_invoice_items_treatment_id ON invoice_items(treatment_id);
+CREATE INDEX idx_medical_histories_patient_id ON medical_histories(patient_id);
```
- [ ] Kindly as requested in the project requirement make sure you have created **index** for all the **FOREIGN KEYS**. You do not have the code for creating the **index** for `medical_history_id`, kindly add it to follow the project requirement.

LionRouge1 reviewed (changes requested):
Hi @RileyManda && @ClarenceAtim, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![almost](https://media.giphy.com/media/KyBesf8kNT6s88SEgb/giphy.gif)

## Highlights
- Use a relational database :heavy_check_mark: 
- Use a diagram given :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ClarenceAtim committed:
Add changes according to request changes

OyePriscilla commented on db_schema_diagram.sql:
```diff
@@ -0,0 +1,70 @@
+    total_amount DECIMAL,
+    generated_at TIMESTAMP,
+    paid_at TIMESTAMP,
+    medical_history_id INTEGER
+);
```
- [ ] As it was raised earlier, kindly create an index for this foreign key: **medical_history_id INTEGER.**

`CREATE INDEX ON invoices (medical_history_id);`

[optional] You did not set the **id** as foreign key, I implore you to implement it. The same is applicable to `medical_histories` table as well. 👍

OyePriscilla reviewed (changes requested):
Hi @RileyManda and @ClarenceAtim 👏 👏, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

![so-close-so-close-seth-meyers](https://user-images.githubusercontent.com/103386183/196425724-8ca5825c-b02a-4983-b976-4bd0a2f70f9a.gif)

 ## Highlights
Implemented `many to many` relationship ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

RileyManda committed:
Implemented index on invoices

RileyManda committed:
Implemented(optional) foreign keys to medical histories

RileyManda commented:
We have completed implementing the **requested changes:** :

- [x] Implemented index on invoices
- [x] Implemented(optional)foreign key on medical histories.

Please kindly review our implementation.

Thank you 🙏🏻

jerryowusu reviewed (approved):
Hi @RileyManda & @ClarenceAtim 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![chelsea-mason-approve](https://github.com/ClarenceAtim/Vet-clinic/assets/65233860/5e219a32-7843-41c3-ad8a-845140569eac)


## Highlights
Well organized code :+1: 
Good effort in making the requested changes :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
